Packs/Rapid7_Nexpose/Integrations/Rapid7_Nexpose/Rapid7_Nexpose.py

```diff
@@ -6458,6 +6725,968 @@ def get_list_asset_group_command(
+    :param data_format: Should only be filled in case the 'events' parameter contains a string of raw
+        events in the format of 'leef' or 'cef'. In other cases the data_format will be set automatically.
+
+    :type url_key: ``str``
+    :param url_key: The param dict key where the integration url is located at. the default is 'url'.
+
+    :type num_of_attempts: ``int``
+    :param num_of_attempts: The num of attempts to do in case there is an api limit (429 error codes)
+
+    :type chunk_size: ``int``
+    :param chunk_size: Advanced - The maximal size of each chunk size we send to API. Limit of 9 MB will be inforced.
+
+    :type should_update_health_module: ``bool``
+    :param should_update_health_module: whether to trigger the health module showing how many events were sent to xsiam
+
+    :type add_proxy_to_request :``bool``
+    :param add_proxy_to_request: whether to add proxy to the send evnets request.
+
+    :type send_events_asynchronously: ``bool``
+    :param send_events_asynchronously: whether to use asyncio to send the events to xsiam asynchronously or not.
+    Note that when set to True, the debug should be done from the integration itself.
+
+    :type data_size_expected_to_split_evenly: ``bool``
+    :param data_size_expected_to_split_evenly: whether the events should be about the same size or not.
+    Use this to split data to chunks faster.
+
+    :return: Either None if running regularly or a list of asyncio task objects if running asynchronously:.
+    In case of running asynchronously:, the list of tasks will hold the number of events sent and can be accessed by:
+    await asyncio.gather(*tasks)
+    :rtype: ``List[Task]`` or ``None``
+    """
+    return rapid7_send_data_to_xsiam(
+        events,
+        vendor,
+        product,
+        data_format,
+        url_key,
+        num_of_attempts,
+        chunk_size,
+        data_type="assets",
+        should_update_health_module=should_update_health_module,
+        add_proxy_to_request=add_proxy_to_request,
+        send_events_asynchronously=send_events_asynchronously,
+        data_size_expected_to_split_evenly=data_size_expected_to_split_evenly,
+        counter=counter,
+        items_count=items_count,
+        snapshot_id=snapshot_id,
+    )
+
+
+async def run_full_collector_workflow(client: InsightVMClient, event_type: str, batch_size: int = 1000):
+    """
+    Orchestrates the full report lifecycle: State Check, Create/Generate, Stream, Cleanup.
+
+    The function signature is adapted to receive the client instance.
+    """
+    integration_context = get_integration_context()
+    demisto.debug(f"Got integration_context: {integration_context} for {event_type}")
+
+    # 1. RETRIEVE STATE
+    event_integration_context = integration_context.get(event_type, {})
+    report_id = event_integration_context.get("report_id", "")
+    instance_id = event_integration_context.get("instance_id", "")
+    finish = event_integration_context.get(
```

Comment:
finish is unclear what we havent finished, have we finished
downloading report?
sending to xsiam
cleanup? 

think we should give a name that more clearly shows what the state is